shell/platform/darwin/ios/ios_surface.mm

```diff
@@ -21,16 +23,14 @@
       switch (context->GetBackend()) {
         case IOSRenderingBackend::kSkia:
           return std::make_unique<IOSSurfaceMetalSkia>(
-              fml::scoped_nsobject<CAMetalLayer>(
-                  reinterpret_cast<CAMetalLayer*>([layer.get() retain])),  // Metal layer
-              std::move(context)                                           // context
+              fml::scoped_nsobject<CAMetalLayer>((CAMetalLayer*)layer.get()),  // Metal layer
```

Comment:
This method is called from `platform_view_ios`
https://github.com/flutter/engine/blob/57a2163d25b7bdb4c739211ec71de643e2b6610b/shell/platform/darwin/ios/platform_view_ios.mm#L121-L122

Reply:
I don't think this is good because .get() has an annotation that the result isn't retained, so it won't be equivalent to what we've had before ( https://github.com/flutter/engine/blob/1363b52813e351414240aef7f15f849bf97f4ca7/fml/platform/darwin/scoped_typeref.h#L123 )

Reply:
What's the concern here? The purpose of the `scoped_nsobject` update from Chromium that cyanglaz did is that we now have a version of `scoped_nsobject` that has all the annotations necessary for ARC to reason about it correctly.